Marlin/src/gcode/config/M200-M205.cpp

```diff
@@ -159,7 +159,7 @@ void GcodeSuite::M205() {
           SERIAL_ECHOLNPGM("WARNING! Low Z Jerk may lead to unwanted pauses.");
       #endif
     }
-    #if !BOTH(JUNCTION_DEVIATION, LIN_ADVANCE)
+    #if HAS_CLASSIC_E_JERK
```

Comment:
I was tracking down why I was unable to adjust E jerk on my delta with linear advance, and traced the issue to this commit. It looks like a boolean logic rearrangement error slipped in here, and a few other spots with the same replacement.
before:
`!(JD && LA) == !JD || !LA`
after:
`HAS_CLASSIC_E_JERK == (HAS_CLASSIC_JERK && DISABLED(LIN_ADVANCE)) == (!JD && !LA) == !(JD || LA)`

Reply:
@sjasonsmith has been doing a lot of work on improving the detla configs. Can you confirm/test this?

Reply:
I've only been messing with probing-related issues. I have never even attempted to use JD or LA, or even adjusted jerk, so I'm not sure how they all are supposed to relate to one another.

Those all sound like things I would need if I actually printed things. I just probe serial lines, flash firmware, and crash print heads into things!

Reply:
Please check this PR. It is not a correct one (probably a mistake when opening it) but the content is 
a) related to a couple of logic things not only in the file @sbelljr referenced but also in a couple of others
and
b) solves my issue https://github.com/MarlinFirmware/Marlin/issues/15908
I tested the solution on my Delta and can confirm it works.

https://github.com/MarlinFirmware/Marlin/pull/15967